ros2_generate_interface_docs/ros2_generate_interface_docs/utils.py

```diff
@@ -0,0 +1,153 @@
+
+
+def get_templates_dir():
+    """Return template directory."""
+    return os.path.join(os.path.dirname(__file__), _TEMPLATES_DIR)
+
+
+def copy_css_style(folder_name):
+    """Copy the css style file in the folder_name."""
+    style_css, _ = load_template('styles.css')
+    with open(os.path.join(folder_name, 'styles.css'), 'w') as f:
+        f.write(style_css)
+
+    msg_style_css, _ = load_template('msg-styles.css')
+    with open(os.path.join(folder_name, 'msg-styles.css'), 'w') as f:
+        f.write(msg_style_css)
+
+
+def load_template(filename):
+    """
+    Look up file within rosdoc ROS package.
+
+    return its content, may sys.exit on error.
+    Contents are cached with filename as key.
+
+    :returns: cached file contents
+    """
+    filename = os.path.join(get_templates_dir(), filename)
+    if not os.path.isfile(filename):
+        sys.stderr.write("Cannot locate template file '%s'\n" % (filename))
+        sys.exit(1)
+    with open(filename, 'r') as f:
+        content = f.read()
+        if not content:
+            raise Exception("Template file '%s' is empty\n" % (filename))
+        return content, filename
+
+
+def write_template(template_name, data, output_file):
+    msg_index_template, template_path = load_template(template_name)
+    output = StringIO()
+    interpreter = em.Interpreter(
+        output=output,
+        options={
+            em.BUFFERED_OPT: True,
+            em.RAW_OPT: True,
+        },
+    )
+    try:
+        with open(template_path, 'r') as h:
+            template_content = h.read()
+            interpreter.invoke(
+                'beforeFile', name=template_name, file=h, locals=data)
+        interpreter.string(template_content, template_path, locals=data)
+        interpreter.invoke('afterFile')
+    except Exception as e:  # noqa: F841
+        if os.path.exists(output_file):
+            os.remove(output_file)
+        print(f"{e.__class__.__name__} when expanding '{template_name}' into "
+              f"'{output_file}': {e}", file=sys.stderr)
+        raise
+
+    content = output.getvalue()
+    interpreter.shutdown()
+
+    with open(os.path.join(output_file), 'w') as f:
+        f.write(content)
+
+
+def generate_compact_definition(imported_interface, indent, get_slot_types):
+    ignored_keys_ = IGNORED_KEYS
+    ignored_keys_ += list(imported_interface.get_fields_and_field_types().keys())
+    ignored_keys_ += ['_'+x for x in imported_interface.get_fields_and_field_types().keys()]
+    compact = {'constant_types': [],
+               'constant_names': [],
+               'links': [],
+               'field_types': [],
+               'field_names': []}
+    for key in imported_interface.__dict__.keys():
+        if key not in ignored_keys_:
+            compact['constant_types'].append(key)
+            compact['constant_names'].append(getattr(imported_interface, key))
+
+    fields = imported_interface.get_fields_and_field_types()
+    for field in fields.keys():
```

Comment:
You use fields[field] in 134 so you can just iterate through the pairs:
```
for fields, field_types in fields.items():
```

Reply:
290fe42ce390275d8255d3705de713609156c03c